### PR TITLE
Update security_system.py

### LINKED ISSUE #5: Implement Loading and Mapping Functions in security_system.py
### Description
In the `security_system.py` file, there are several sections marked with `# TODO 1` that require implementation. These tasks involve loading a trained face recognition model, specifying the path to a custom Haar Cascade classifier, loading authorized persons from a CSV file, and loading label-to-name mapping from another CSV file.

### Expected behavior
- Implement the code within the `# TODO 1` sections in `security_system.py` to:
  - Load the trained face recognition model.
  - Specify the path to the custom Haar Cascade classifier.
  - Load authorized persons from a CSV file.
  - Load label-to-name mapping from a CSV file.

### Additional context
- The code structure and comments are already provided in the file.
- You can locate the specific sections that need implementation by searching for `# TODO 1` in the `security_system.py` file.

Completing these tasks will ensure that the security system has access to the necessary data and models to perform face recognition and person identification accurately.

### PR DESCRIPTION
**Pull request by Aqua-16**
Updated over the previous version.

## Description

Loaded the model, specified the cascade paths, implemented the dictionary functions as well as function responsible for getting names. Used those functions to finish the rest of the work.

Fixes #5 : Updated the previous issues as follows:-

1. Fixed typos
2. Since `model_path` and `cascade_path` are optional arguments, they are only given a specified path if no arguments have 
been passed to them. This will allow the code to use user-define classifiers and models.

The following has been implemented in general:-

1. The `authorized_persons.csv` file is being read and stored in a dictionary as key : value pairs of Name and Authorization Status
2. The `label_to_name.csv` file is being read and stored in a dictionary as key : value pairs of Label and Name
3. `get_person_name` file has been implemented to return "Unknown" if confidence score is too low or the provided label is unavailable. Otherwise, it now returns the name associated with the label.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

I have tested this on Jupyter Labs as well as Command Terminal using a Virtual Environment. Each function has also been individually ran to test for any errors.

## Checklist:

- [x] I have registered myself at [Contrihub website](https://sac.mnnit.ac.in/contrihub).
- [x] My code follows the code style of this project.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: app/security_system.py [hunk 1]
```diff
@@ -15,15 +15,16 @@ def __init__(self, algorithm='LBPH', model_path=None, cascade_path=None):
...
 
-        # TODO 1
+        # TODO 1 : Done
         # Load the trained face recognition model
-
+        model_path = "trained_models/trained_face_model_LBPH.xml"
+        self.face_recognizer.read(model_path)
         # Specify the path to the custom Haar Cascade classifier
-
+        cascade_path = "haar_face.xml"
```
Comment (sanskaromar) [inline app/security_system.py]: Did you notice `cascade_path` is an optional parameter? Are you sure your code is the best way to handle that?

File: app/security_system.py [hunk 2]
```diff
@@ -15,15 +15,16 @@ def __init__(self, algorithm='LBPH', model_path=None, cascade_path=None):
         else:
             raise ValueError("Invalid algorithm specified")
 
-        # TODO 1
+        # TODO 1 : Done
         # Load the trained face recognition model
-
+        model_path = "trained_models/trained_face_model_LBPH.xml"
```
Comment (sanskaromar) [inline app/security_system.py]: Why hard coding `model_path`?

File: app/security_system.py [hunk 3]
```diff
@@ -15,15 +15,16 @@ def __init__(self, algorithm='LBPH', model_path=None, cascade_path=None):
...
+        self.face_recognizer.read(model_path)
         # Specify the path to the custom Haar Cascade classifier
-
+        cascade_path = "haar_face.xml"
         # Load authorized persons from CSV
-        
+        self.authorizer_persons = self.load_authorized_persons()
```
Comment (sanskaromar) [inline app/security_system.py]: A typo, I presume? (`authorizer`)

**Review by sanskaromar — CHANGES_REQUESTED** [review]
Please review your code and make the required changes. You may also wanna analyse rest of your code.

**Comment by sanskaromar** [comment]
@Aqua-16 I'd appreciate it if the commit messages were clearer about what's being changed. There is no need to mention about `#TODO` there. And rename your PR to convention `Fixes #(issue): Description`.

**Comment by Aqua-16** [comment]
Thank you for bringing this to my attention sir, I'll fix it right away

**Comment by sanskaromar** [comment]
Are you still working on this? Btw I would encourage a better commit message.

You can refer this [resource](https://www.freecodecamp.org/news/how-to-write-better-git-commit-messages/) on how to write better commit messages.

**Comment by Aqua-16** [comment]
Yes Sir I was still working on it, but I think I'm done now. I have also updated the description to reflect my overall changes.

File: app/security_system.py [hunk 4]
```diff
@@ -17,13 +17,16 @@ def __init__(self, algorithm='LBPH', model_path=None, cascade_path=None):
...
         # TODO 1
         # Load the trained face recognition model
-
+        if(model_path==None):
+            self.model_path = "trained_models/trained_face_model_LBPH.xml"
+        self.face_recognizer.read(self.model_path)
         # Specify the path to the custom Haar Cascade classifier
-
+        if(cascade_path==None):
+            self.cascade_path = "haar_face.xml"
```
Comment (sanskaromar) [inline app/security_system.py:25]: Should've tried to follow projects path convention for files. For example here it should've been `app/haar_face.xml`.

**Review by sanskaromar — APPROVED** [review]
Approving these changes. We can do code refactoring in a separate issue later.


**Please write better commit messages**

File: app/security_system.py (hunk 4 above)
Comment (Aqua-16) [reply app/security_system.py:25]: Understood Sir. I'll read up on Commit messages to write better next time. Thank you for the review.

**Pull request merged**